ApollosProject/apollos-prototype
# Front-load Photo Permissions

## DESCRIPTION

### What does this PR do, or why is it needed?

Makes the app ask for photo library permissions as soon as you tap the button to change your avatar. This makes it so we can avoid an issue where asking for permission after taking a picture causes the picture not to change.

This is not an issue on android.

### What design trade-offs/decisions were made?

We added a new dependency so that we can do this: RNPermissions; it's pretty useful, though, so we may be able to use it elsewhere.

### How do I test this PR?

🚨You may need to clean your project with `npx react-native-clean-project --remove-iOS-build --remove-iOS-pods` before the button will work properly. I had to. PS. You'll need to build to a device to test this.

1. Go to the user-settings screen
2. Tap the camera button on the avatar
3. Grant permission for the photo library
4. Select 'Take Picture'
5. Grant permission for the camera
6. Take a picture
7. Your picture should change

### What automated tests did you write?

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
